app/models/partner.rb

```diff
@@ -60,6 +62,10 @@ class Partner < ApplicationRecord
     where(status: status.to_sym)
   }
 
+  def invite_new_partner
+    PartnerUser.invite!(email: email, partner: profile)
+  end
```

Comment:
Could we make this into a private method since it is only going to be called in a callback? 

I believe the choice to make it public was to make it testable. However, the test you added doesn't test the callback but just tests that the public method works. Could you instead move this to private and make sure that when you change the email via an update, that the `PartnerUser` method gets called with an invite on the right parameters?